File: app/src/main/java/com/kota/Bahamut/Command/BahamutCommandPushArticle.java
package com.kota.Bahamut.Command;

import androidx.annotation.NonNull;

import com.kota.Bahamut.ListPage.TelnetListPage;
import com.kota.Bahamut.ListPage.TelnetListPageBlock;
import com.kota.Telnet.TelnetClient;

public class BahamutCommandPushArticle extends TelnetCommand {
    int _article_index;

    public BahamutCommandPushArticle(int articleIndex) {
        this._article_index = articleIndex;
        this.Action = PushArticle;
    }

    public void execute(TelnetListPage aListPage) {
        if (this._article_index > 0) {
            TelnetClient.getClient().sendStringToServer(this._article_index + "\ngx\n");
        }
    }

    public void executeFinished(TelnetListPage aListPage, TelnetListPageBlock aPageData) {
        setDone(true);
    }

    @NonNull
    public String toString() {
        return "[PushArticle][articleIndex=" + this._article_index + "]";
    }
}

File: app/src/main/java/com/kota/Bahamut/command/BahamutCommandPushArticle.kt
package com.kota.Bahamut.command

import com.kota.Bahamut.listPage.TelnetListPage
import com.kota.Bahamut.listPage.TelnetListPageBlock
import com.kota.telnet.TelnetClient

class BahamutCommandPushArticle(var articleIndex: Int) : TelnetCommand() {
    init {
        this.action = BahamutCommandDef.Companion.PUSH_ARTICLE
    }

    override fun execute(telnetListPage: TelnetListPage) {
        if (this.articleIndex > 0) {
            TelnetClient.client!!.sendStringToServer(this.articleIndex.toString() + "\ngx")
        }
    }

    override fun executeFinished(telnetListPage: TelnetListPage, telnetListPageBlock: TelnetListPageBlock) {
        isDone = true
    }

    override fun toString(): String {
        return "[PushArticle][articleIndex=" + this.articleIndex + "]"
    }
}
